CorsixTH/Lua/hospitals/player_hospital.lua

```diff
@@ -192,29 +198,126 @@ function PlayerHospital:monthlyAdviceChecks()
+
+  elseif current_month == 3 then
+    self:giveAdvice({_A.warnings.no_desk}, 1, true)
+
+  elseif current_month == 8 then
+    self:giveAdvice({_A.warnings.no_desk_1}, 1, true)
+
+  elseif current_month == 11 then
+    if self.visitors == 0 then
+      self:giveAdvice({_A.warnings.no_desk_2}, 1, true)
+    else
+      self:giveAdvice({_A.warnings.no_desk_3}, 1, true)
+    end
+  end
+end
+
+--! Give advice to the user about having bought a reception desk.
+function PlayerHospital:msgReceptionDesk()
+  local num_receptionists = self:countStaffOfCategory("Receptionist")
+
+  if not self.world.ui.start_tutorial and num_receptionists == 0 then
+    self:giveAdvice({_A.room_requirements.reception_need_receptionist})
+  elseif num_receptionists > 0 and self:countReceptionDesks() == 1 and
+      not self.adviser_data.reception_advice and self.world:date():monthOfGame() > 3 then
+    self:giveAdvice({_A.warnings.no_desk_5})
+    self.adviser_data.reception_advice = true
+  end
+end
+
+--! Give advice to the user about maintenance of plants.
+function PlayerHospital:msgPlant()
+  local num_handyman = self:countStaffOfCategory("Handyman")
+
+  if num_handyman == 0 then
+    self:giveAdvice({_A.staff_advice.need_handyman_plants})
+  end
+end
+
+--! Show the 'Gates to hell' animation.
+--!param entity (Entity) Gates to hell.
+function PlayerHospital:showGatesToHell(entity)
+  local anim_func = --[[persistable:lava_hole_spawn_animation_end]]
+    function(anim_entity)
+      anim_entity:setAnimation(1602)
+    end
+
+  entity:playEntitySounds("LAVA00*.WAV", {0,1350,1150,950,750,350},
+      {0,1450,1250,1050,850,450}, 40)
+  entity:setTimer(entity.world:getAnimLength(2550), anim_func)
+  entity:setAnimation(2550)
 end
 
 --! Advises the player.
 --!param msgs (array of string) Messages to select from.
 --!param rnd_frac (optional float in range (0, 1]) Fraction of times that the
 --    call actually says something.
-function PlayerHospital:giveAdvice(msgs, rnd_frac)
+--!param stay_up (bool) If true, let the adviser remain visible afterwards.
+function PlayerHospital:giveAdvice(msgs, rnd_frac, stay_up)
   local max_rnd = #msgs
   if rnd_frac and rnd_frac > 0 and rnd_frac < 1 then
     -- Scale by the fraction.
     max_rnd = math.floor(max_rnd / rnd_frac)
   end
 
   local index = (max_rnd == 1) and 1 or math.random(1, max_rnd)
-  if index <= #msgs then self.world.ui.adviser:say(msgs[index]) end
+  if index <= #msgs then self.world.ui.adviser:say(msgs[index], stay_up) end
+end
+
+-- Update the hospital each tick.
+function PlayerHospital:tick()
+  -- Add some random background sounds, ringing phones, coughing, belching etc.
+  if math.random(1, 100) == 3 then
+    -- Wait until there are some patients in the hospital and a room, otherwise you
+    -- will wonder who is coughing or who is the receptionist telephoning!
+    -- Opted for gp as you can't run the hospital without one.
+    if self:countRoomOfType("gp") > 0 and self:countPatients(3) > 2 then
```

Comment:
This sounds stuff doesn't belong here - I'm not quite sure where it belongs, but (as far as I can remember) you can hear the environmental sounds of other hospitals in multiplayer.

Reply:
Moved the code back while keeping the perfoamnce improvements, and added a TODO multiplayer note.